### PR TITLE
feat(caniuse): group browser versions with similar support

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/marcoscaceres/respec.org/issues/149

We should update the styles in respec (partial support, behind flag are shown as same, but title are different).
Also, titles in HTML here have some room for improvement.
~And as a performance improvement, we should do the grouping in scraper (while still supporting legacy json format).~ done.

<details>
<summary>Demo: Payment Request</summary>

image
</details>

<details>
<summary>Demo: Manifest</summary>

image
</details>

<details>
<summary>Generated HTML for web-app-manifest</summary>

```html
<div class="caniuse-browser">
  <button class="caniuse-cell y" title="Supported.">Chrome 91</button>
  <ul>
    <li class="caniuse-cell y" title="Supported.">39-90</li>
    <li class="caniuse-cell n" title="Not supported by default, but can be enabled.">4-38</li>
  </ul>
</div>
<div class="caniuse-browser">
  <button class="caniuse-cell n d" title="Not supported by default, but can be enabled. Disabled by default (needs to enabled).">Firefox 87</button>
  <ul>
    <li class="caniuse-cell n d" title="Not supported by default, but can be enabled. Disabled by default (needs to enabled).">76-86</li>
    <li class="caniuse-cell n" title="Not supported by default, but can be enabled.">2-75</li>
  </ul>
</div>
<div class="caniuse-browser">
  <button class="caniuse-cell n" title="Not supported by default, but can be enabled.">Safari TP</button>
  <ul>
    <li class="caniuse-cell n" title="Not supported by default, but can be enabled.">3.1-14</li>
  </ul>
</div>
<div class="caniuse-browser">
  <button class="caniuse-cell y" title="Supported.">Edge 88</button>
  <ul>
    <li class="caniuse-cell y" title="Supported.">79-87</li>
    <li class="caniuse-cell n" title="Not supported by default, but can be enabled.">12-18</li>
  </ul>
</div>
<a href="https://caniuse.com/web-app-manifest">More info</a>
```

</details>

### CODE REVIEW

**Comment by sidvishnoi** [comment]
| Before | After |
| --- | --- |
| image | image |

**Review by marcoscaceres — APPROVED** [review]
Looks good. Will be quite useful!

File: routes/caniuse/lib/index.ts [hunk 1]
```diff
@@ -56,24 +53,38 @@ export async function createResponseBodyJSON(options: NormalizedOptions) {
...
-  const data = await createResponseBodyJSON(options);
-  return data === null ? null : formatAsHTML(options, data);
+  const { feature, browsers } = options;
+  const allData = await getData(feature);
+  if (!allData) {
+    return null;
+  }
+
+  if (!browsers.length) {
+    browsers.push(...Object.keys(allData));
```
Comment (sidvishnoi) [inline routes/caniuse/lib/index.ts:72]:
```suggestion
    browsers.push(...Object.keys(allData.summary));
```

File: routes/caniuse/lib/index.ts [hunk 2]
```diff
@@ -56,24 +53,38 @@ export async function createResponseBodyJSON(options: NormalizedOptions) {
     browsers.push(...Object.keys(data));
```
Comment (sidvishnoi) [inline routes/caniuse/lib/index.ts:53]:
```suggestion
    browsers.push(...Object.keys(data.all));
```

File: routes/caniuse/lib/index.ts (hunk 1 above)
Comment (sidvishnoi) [reply routes/caniuse/lib/index.ts:72]: TypeScript doesn't save when types are same 😄

**Pull request merged**